nephio-ansible-install/roles/gitea/create/tasks/main.yaml

```diff
@@ -0,0 +1,57 @@
+---
+- name: Get UID
+  getent:
+    database: passwd
+  when:
+    - gitea_username is defined
+    - gitea_password is defined
+- name: Create a gitea directory
+  ansible.builtin.file:
+    path: /home/{{ cloud_user}}/gitea
+    state: directory
+    mode: '0755'
+- name: Start gitea container
+  docker_container:
+    name: gitea
+    image: gitea/gitea:1.17.3-rootless
+    networks:
+      - name: kind
+    user: "{{ getent_passwd[cloud_user].1 }}:{{ getent_passwd[cloud_user].2 }}"
+    restart: yes
+    ports:
+      - "127.0.0.1:3000:3000/tcp"
+      - "127.0.0.1:2222:2222/tcp"
+    env:
+        GITEA_APP_INI: "/var/lib/gitea/custom/conf/app.ini"
+        INSTALL_LOCK: "true"
+    volumes:
+      - /home/{{ cloud_user }}/gitea:/var/lib/gitea:rw
+  when:
+    - gitea_username is defined
+    - gitea_password is defined
+- name: Wait for gitea to initialise
+  ansible.builtin.wait_for:
+    path: /home/{{ cloud_user }}/gitea/ssh
+  when:
+    - gitea_username is defined
+    - gitea_password is defined
+- name: Creating nephio user in gitea
```

Comment:
FYI Ran into the below when this task was getting executed

```
TASK [gitea/create : Creating nephio user in gitea] *********************************************************************************************************************************************************************************************
fatal: [localhost]: FAILED! => {"changed": true, "cmd": "docker exec -it gitea  gitea admin user create    --username nephio    --password nephio    --must-change-password=false    --access-token    --email=nephio@nephio.org > /home/vjayaram/gitea/nephio-create\n", "delta": "0:00:00.034046", "end": "2022-12-13 03:29:52.192736", "msg": "non-zero return code", "rc": 1, "start": "2022-12-13 03:29:52.158690", "stderr": "the input device is not a TTY", "stderr_lines": ["the input device is not a TTY"], "stdout": "", "stdout_lines": []}
```

Reply:
fixed: [5c9d912](https://github.com/nephio-project/one-summit-22-workshop/pull/106/commits/5c9d912f784f71a0de58bbea88ccc54bbbbb9cfe)